### PR TITLE
Add admin user role and policies

### PR DESCRIPTION
**Pull request by Kirollos-Hanna**

### CODE REVIEW

File: src/app/Providers/AuthServiceProvider.php [hunk 1]
```diff
@@ -16,6 +20,8 @@ class AuthServiceProvider extends ServiceProvider
      */
     protected $policies = [
         'App\Models\Model' => 'App\Policies\ModelPolicy',
+        User::class => AdminPolicy::class,
```
Comment (Mhany95) [inline src/app/Providers/AuthServiceProvider.php]: Create a separate policy for each model, policy name should follow the mode's name

File: src/app/Policies/AdminPolicy.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Policies;
+
+use App\Models\User;
+use Illuminate\Auth\Access\HandlesAuthorization;
+use Bouncer;
+
+class AdminPolicy
```
Comment (Mhany95) [inline src/app/Policies/AdminPolicy.php]: Policy should contain all the methods

File: src/routes/web.php [hunk 3]
```diff
@@ -49,4 +49,8 @@
     return view('welcome');
 });
 
+Route::group(['middleware' =>['role:admin']], function(){
```
Comment (Mhany95) [inline src/routes/web.php]: ?

File: src/database/seeders/BouncerSeeder.php [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+     * Run the database seeds.
+     *
+     * @return void
+     */
+    public function run()
+    {
+        // $adminUser = User::findOrFail(3);
+        // Log::info('User failed to login.', ['id' => $adminUser->name]);
+        // $adminUser->assign('admin');
+        Bouncer::allow('admin')->to('access-resources');
```
Comment (Mhany95) [inline src/database/seeders/BouncerSeeder.php]: no one is assigned the "admin" role?

File: src/app/Providers/AuthServiceProvider.php [hunk 5]
```diff
@@ -27,6 +33,10 @@ public function boot()
     {
         $this->registerPolicies();
 
+        Gate::define('access-resources', function ($user) {
```
Comment (Mhany95) [inline src/app/Providers/AuthServiceProvider.php]: ?

File: .vscode/settings.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+    "compile-hero.disable-compile-files-on-did-save-code": true
```
Comment (abdelrahmanna) [inline .vscode/settings.json:2]: add this folder to .gitignore

**Pull request merged**